mono/metadata/cominterop.c

```diff
@@ -1884,26 +1879,22 @@ ves_icall_Mono_Interop_ComInteropProxy_FindProxy (gpointer pUnk, MonoError *erro
  *
  * Returns: the corresponding object for the CCW
  */
-static MonoObject*
-cominterop_get_ccw_object (MonoCCWInterface* ccw_entry, gboolean verify)
+static MonoObjectHandle
+cominterop_get_ccw_object_handle (MonoCCWInterface* ccw_entry, gboolean verify)
 {
-	MonoCCW *ccw = NULL;
-
 	/* no CCW's exist yet */
 	if (!ccw_interface_hash)
-		return NULL;
+		return NULL_HANDLE;
 
-	if (verify) {
-		ccw = (MonoCCW *)g_hash_table_lookup (ccw_interface_hash, ccw_entry);
-	}
-	else {
-		ccw = ccw_entry->ccw;
-		g_assert (ccw);
-	}
-	if (ccw)
-		return mono_gchandle_get_target (ccw->gc_handle);
-	else
-		return NULL;
+	MonoCCW * const ccw = verify ? (MonoCCW *)g_hash_table_lookup (ccw_interface_hash, ccw_entry) : ccw_entry->ccw;
+	g_assert (verify || ccw);
+	return ccw ? mono_gchandle_get_target_handle (ccw->gc_handle) : NULL_HANDLE;
+}
+
+static MonoObject*
+cominterop_get_ccw_object (MonoCCWInterface* ccw_entry, gboolean verify)
+{
+	return MONO_HANDLE_RAW (cominterop_get_ccw_object_handle (ccw_entry, verify));
```

Comment:
This is going to leak handles.   Note that this function is (sometimes, but not always) used as a dynamically-registered icall:
https://github.com/mono/mono/blob/master/mono/metadata/cominterop.c#L634

So I would suggest:
1. Make a new entrypoint `static ves_icall_cominterop_get_ccw_object` and use that exclusively as a dynamically registered icall.
2. Keep the `cominterop_get_ccw_object` for now until its remaining caller(s) are comverted to use the handle-ized version.
3. In both (1) and (2) add handle frame entry/exit macros so that it doesn't leak the coop handles.